packages/video_player/video_player/example/lib/main.dart

```diff
@@ -10,461 +10,303 @@
 import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
 import 'package:video_player/video_player.dart';
-import 'package:flutter/scheduler.dart';
 
-/// Controls play and pause of [controller].
-///
-/// Toggles play/pause on tap (accompanied by a fading status icon).
-///
-/// Plays (looping) on initialization, and mutes on deactivation.
-class VideoPlayPause extends StatefulWidget {
-  VideoPlayPause(this.controller);
-
-  final VideoPlayerController controller;
+void main() {
```

Comment:
The diff for this is very difficult to follow, could you include before/after screenshots of what the example looked like?

Reply:
It's mostly a refactor, I was able to reduce the number of lines of code, and (subjectively) make the example easier to follow by taking out the inheritance. 

Part of it was a bug fix (by changing the play pause button): https://github.com/flutter/flutter/issues/49813

And finally, the only real visual change is I added closed captions to the video on the first page of them example app.

Reply:
It's not a blocker here, but FWIW patches like this would be easier to review in the future if the refactor was split into a separate patch. Generally I prefer landing refactors separately, that way it's obvious which lines are meant to introduce changes and if there's a bug with the new functionality the refactor doesn't need to be rolled back too. Thank you for cleaning up this code though! Appreciate it. I manually tested this and saw no visual changes to the example app other than the extra captions.

Reply:
Yeah sorry about that, there were a couple of bugs that were keeping me from testing properly so I did the refactor + bug fixes in one fell swoop. The only other thing I changed was the play/pause overlay (the button before wasn't working and was causing bugs).